src/crypto/tls/ech.go

```diff
@@ -584,6 +588,18 @@ func echEncodeClientHelloInner(innerData []byte) []byte {
 		panic(err) // Host encountered internal error
 	}
 
+	// Add padding.
+	paddingLen := 0
+	if serverNameLen > 0 {
+		paddingLen += max(0, maxNameLen-serverNameLen)
+	} else {
+		paddingLen += maxNameLen + 9
```

Comment:
Is this '9' an arbitrary number? if it is not, it might be good to make it a constant.

Reply:
Done. I also added some documentation to the spec to explain why `paddingLen` is calculated this way.